### PR TITLE
Feature/107 refboxfeatures

### PR DESCRIPTION
**Pull request by gwvdh**

### CODE REVIEW

File: front/src/RefboxSettings/ConfigDropdown.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+/**
+ * Constructs two dropdowns for the type of game and role
+ * Author: G.W. van der Heijden
+ * Author: S.A. Tanja
+ * Author: T.T.P. Franken
+ *
+ * Props:
+ * - value: current value of the dropdown
+ * - onChange: a function on what to do when the button is pressed
+ */
```
[the remark this answers is not in the document]
Comment (gwvdh) [reply front/src/RefboxSettings/ConfigDropdown.js]: fixed

File: front/src/RefboxField/index.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          grid-template-columns: repeat(2, 1fr);
+        `};
+  margin: auto;
+  width: 16rem;
+  float: center;
+`;
+
+const tags = props => {
+  if (props.isPenalty) {
+    return ["Soft", "Medium", "Hard"];
```
Comment (gwvdh) [inline front/src/RefboxField/index.js]: This part is not tested

File: front/src/RefboxField/index.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  } else {
+    return ["KO", "FK", "GK", "TI", "C", "P"];
+  }
+};
+
+RefboxField.propType = {
+  isPenalty: PropTypes.bool.isRequired,
+  teamColor: PropTypes.oneOf(["cyan", "magenta"]).isRequired
```
Comment (gwvdh) [inline front/src/RefboxField/index.js]: Add onClick function

File: front/src/RefboxField/index.test.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import RefboxField from ".";
+import renderer from "react-test-renderer";
+
+describe("RefboxField", () => {
+  it("shows a cyan refbox field with all buttons", () => {
+    const component = renderer.create(
+      <RefboxField teamColor="cyan" onClick={(tag, color) => {}} />
```
Comment (gwvdh) [inline front/src/RefboxField/index.test.js]: Add isPenalty argument

File: front/src/Turtle/__snapshots__/index.test.js.snap [hunk 5]
```diff
@@ -30,14 +30,15 @@ exports[`Turtle should match snapshot 1`] = `
     className="sc-bwzfXH lopfab"
   >
     <p>
-      Turtle 
+      Turtle
```
Comment (gwvdh) [inline front/src/Turtle/__snapshots__/index.test.js.snap]: Linting does this and thus consequently breaks the test

File: front/src/RefboxSettings/index.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+ * - Go in
+ * - Go out
+ * - Selectors (Match selector & Role / demo selector), see ConfigDropdown.js
+ *
+ * Author: G.W. van der Heijden
+ * Author: S.A. Tanja
+ * Author: T.T.P. Franken
+ *
+ * Props:
+ * - dbOnClick: a function on what to do when DropBall is pressed
```
Comment (helaan) [inline front/src/RefboxSettings/index.js]: Most event handlers start with on, so these don't follow this convention. Why? You may also want to consolidate these handlers, see turtle for that.

File: front/src/RefboxField/RefboxButton.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+ * A button part of the refbox. Does a callback to its parent to handle onClick events
+ * Author: S.A. Tanja
+ * Author: G.W. van der Heijden
+ *
+ * Props:
+ *  - teamColor: the team color of the button (cyan or magenta).
+ *  - onClick: a function on what to do when the button is pressed
+ *  - children: the children of the button
+ */
+const RefboxButton = props => {
```
Comment (helaan) [inline front/src/RefboxField/RefboxButton.js]: Why do we need yet another button class? Can we not reuse TurtleEnableButton or something?

Comment (gwvdh) [reply front/src/RefboxField/RefboxButton.js]: Because this button is not a turtle button. You could make a individual button class like dropdown.

**Comment by bafonins** [comment]
rebase and you are good ⚡️

**Pull request merged**